model-api-gen-runtime/src/commonMain/kotlin/org/modelix/metamodel/IPropertyValueSerializer.kt

```diff
@@ -78,23 +78,32 @@ object OptionalIntPropertySerializer : IPropertyValueSerializer<Int?> {
     }
 }
 
-class MandatoryEnumSerializer<E : Enum<*>>(private val valueOf: (String?) -> E) : IPropertyValueSerializer<E> {
+class MandatoryEnumSerializer<E : Enum<*>>(
+    private val memberIdOf: (E) -> String,
+    private val fromString: (String?) -> E
+) : IPropertyValueSerializer<E> {
+
     override fun serialize(value: E): String {
-        return value.name
+        return memberIdOf(value) + "/" + value.name
     }
 
     override fun deserialize(serialized: String?): E {
-        return valueOf(serialized)
+        val id = serialized?.substringBefore('/')
+        return fromString(id)
     }
 }
 
-class OptionalEnumSerializer<E : Enum<*>>(private val valueOf: (String?) -> E?) : IPropertyValueSerializer<Enum<*>?> {
-    override fun serialize(value: Enum<*>?): String? {
-        return value?.name
+class OptionalEnumSerializer<E : Enum<*>>(
+    private val memberIdOf: (E) -> String,
+    private val fromString: (String) -> E
+) : IPropertyValueSerializer<E?> {
+
+    override fun serialize(value: E?): String? {
+        return value?.let { memberIdOf(it) + "/" + it.name}
     }
 
-    override fun deserialize(serialized: String?): Enum<*>? {
-        return serialized?.let { valueOf(serialized) }
+    override fun deserialize(serialized: String?): E? {
+        return serialized?.let { fromString(it) }
```

Comment:
The .substringBefore part is missing here. Make OptionalEnumSerializer a subclass of MandatoryEnumSerializer to remove the code duplication. Also rename `fromString` to make clear that it expects the memberId.